test/mongoConnectionStrings.test.ts

```diff
@@ -5,55 +5,80 @@
-        testDatabaseFromConnectionString(`mongodb://router1.example.com:27017,router2.example2.com:27017,router3.example3.com:27017/I like spaces`, "I like spaces");
+        testItemsFromConnectionString(`mongodb://router1.example.com:27017,router2.example2.com:27017,router3.example3.com:27017/abc.def`, ['mongodb', 'router1.example.com:27017,router2.example2.com:27017,router3.example3.com:27017', 'abc.def', undefined]);
+        testItemsFromConnectionString(`mongodb://router1.example.com:27017,router2.example2.com:27017,router3.example3.com:27017/abc.def-ghi_jkl`, ['mongodb', 'router1.example.com:27017,router2.example2.com:27017,router3.example3.com:27017', 'abc.def-ghi_jkl', undefined]);
+        testItemsFromConnectionString(`mongodb://router1.example.com:27017,router2.example2.com:27017,router3.example3.com:27017/I like spaces`, ['mongodb', 'router1.example.com:27017,router2.example2.com:27017,router3.example3.com:27017', 'I like spaces', undefined]);
 
         // emulator: mongodb://localhost:C2y6yDjf5/R+ob0N8A7Cgv30VRDJIWEHLM+4QDU5DE2nQ9nDuVTqobD4b8mGGyPMbIZnqyMsEcaGQy67XIw/Jw==@localhost:10255?ssl=true
     });
+
+    test('addDatabaseToConnectionString', () => {
+        testDatabaseToConnectionString(`mongodb://a.b.c/?readPreference=secondary&maxStalenessSeconds=1201`, 'realdatabase', 'mongodb://a.b.c/realdatabase?readPreference=secondary&maxStalenessSeconds=1201');
+        testDatabaseToConnectionString(`mongodb://my-mongo:ayO83FFfUoHE97Jm7WbfnpNCqiF0Yq0za2YmvuLAKYJKf7h7hQaRKWfZfsv8Ux41H66Gls7lVPEKlKm0ueSozg==@your-mongo.documents.azure.com:10255/our-mongo?ssl=true&replicaSet=globaldb`, 'somedatabase', 'mongodb://my-mongo:ayO83FFfUoHE97Jm7WbfnpNCqiF0Yq0za2YmvuLAKYJKf7h7hQaRKWfZfsv8Ux41H66Gls7lVPEKlKm0ueSozg==@your-mongo.documents.azure.com:10255/our-mongo?ssl=true&replicaSet=globaldb');
+        testDatabaseToConnectionString(`mongodb://my-mongo:ayO83FFfUoHE97Jm7WbfnpNCqiF0Yq0za2YmvuLAKYJKf7h7hQaRKWfZfsv8Ux41H66Gls7lVPEKlKm0ueSozg==@your-mongo.documents.azure.com:10255/our-mongo?ssl=true&replicaSet=globaldb`, 'our-mongo', 'mongodb://my-mongo:ayO83FFfUoHE97Jm7WbfnpNCqiF0Yq0za2YmvuLAKYJKf7h7hQaRKWfZfsv8Ux41H66Gls7lVPEKlKm0ueSozg==@your-mongo.documents.azure.com:10255/our-mongo?ssl=true&replicaSet=globaldb')
+
+        testDatabaseToConnectionString(`mongodb://dbuser:dbpassword@dbname.mlab.com:14118`, 'mydata', 'mongodb://dbuser:dbpassword@dbname.mlab.com:14118/mydata');
+        testDatabaseToConnectionString(`mongodb://dbuser:dbpassword@dbname.mlab.com:14118/`, 'database', 'mongodb://dbuser:dbpassword@dbname.mlab.com:14118/database');
+        testDatabaseToConnectionString(`mongodb+srv://dbuser:dbpassword@dbname.mlab.com:14118/database`, 'database', 'mongodb+srv://dbuser:dbpassword@dbname.mlab.com:14118/database');
+
+        testDatabaseToConnectionString(`mongodb://db1.example.net:27017,db2.example.net:2500/?replicaSet=test`, 'my-database', 'mongodb://db1.example.net:27017,db2.example.net:2500/my-database?replicaSet=test');
```

Comment:
These tests need to change the database name to something _other_ than 'my-database' (which is already in the connection string)

Reply:
Not sure that it's even possible to get parameters like these: database name and database name inside connection string are different. How you can see, for now, I don't replace the name of database in connection string. Is this a way how we want to resolve this?